app/controllers/lead_providers/confirm_schools_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class ConfirmSchoolsController < ApplicationController
+    before_action :load_form
+
+    def show
+      @schools = School.where(id: @confirm_schools_form.school_ids).order(:name).includes(:local_authority)
+      @delivery_partner = DeliveryPartner.find(@confirm_schools_form.delivery_partner_id)
+    end
+
+    def remove
+      school_id = params[:remove][:school_id]
+      @confirm_schools_form.school_ids.delete(school_id)
```

Comment:
Do we need to save this back to the session?

Reply:
@ethanmills Yes and no. :)

Yes, as it should be just a good practice to do this and would protect us from any changes.

No, as `@confirm_schools_form.school_ids` and `session[:confirm_schools_form][:school_ids]` both references the same object, so any modification to `@confirm_schools_form` modifies session as well. This is a bit too magical for my liking and would much rather prefer having explicit session update but I've let it slide for now. If we were to change this line to `@confirm_schools_form.school_ids -= [schoolId]` or sth similar, this would introduce behaviour change - it is covered with the test though.

To do it properly, I think we should stop using `Form.new(session[:key_to_session_hash]` or to modify our form objects to assign mutable objects with `dup`

Reply:
Oh weird.... Ok, I didn't know references to the session worked like that! I agree `dup` is probably the way forward. I much prefer things to be immutable